lib/active_record/connection_adapters/oracle_enhanced_adapter.rb

```diff
@@ -569,6 +569,10 @@ def materialized_views #:nodoc:
         select_values("SELECT LOWER(mview_name) FROM all_mviews WHERE owner = SYS_CONTEXT('userenv', 'current_schema')")
       end
 
+      def synonym_names #:nodoc:
+        select_values("SELECT LOWER(synonym_name) FROM all_synonyms where owner = SYS_CONTEXT('userenv', 'current_schema')")
```

Comment:
I prefer this method name `synonyms` and found changing this method name to `synonyms` causes this error at Schema dumper.

```
NoMethodError: undefined method `name' for "test_synonym":String
```
I'd like to address this error and use `synonyms` method name here to have better consistency with `tables` and `views`.